diff --git a/files/powermenu/type-1/powermenu.sh b/files/powermenu/type-1/powermenu.sh
index dd978af..bc7e4f8 100755
--- a/powermenu/type-1/powermenu.sh
+++ b/powermenu/type-1/powermenu.sh
@@ -70,15 +70,7 @@ run_cmd() {
 			amixer set Master mute
 			systemctl suspend
 		elif [[ $1 == '--logout' ]]; then
-			if [[ "$DESKTOP_SESSION" == 'openbox' ]]; then
-				openbox --exit
-			elif [[ "$DESKTOP_SESSION" == 'bspwm' ]]; then
-				bspc quit
-			elif [[ "$DESKTOP_SESSION" == 'i3' ]]; then
-				i3-msg exit
-			elif [[ "$DESKTOP_SESSION" == 'plasma' ]]; then
-				qdbus org.kde.ksmserver /KSMServer logout 0 0 0
-			fi
+			exit_cmd
 		fi
 	else
 		exit 0
@@ -95,11 +87,7 @@ case ${chosen} in
 		run_cmd --reboot
         ;;
     $lock)
-		if [[ -x '/usr/bin/betterlockscreen' ]]; then
-			betterlockscreen -l
-		elif [[ -x '/usr/bin/i3lock' ]]; then
-			i3lock
-		fi
+		lock_cmd
         ;;
     $suspend)
 		run_cmd --suspend
